config/master/calico-operator.yaml

```diff
@@ -4942,6 +4942,14 @@ rules:
       - update
       - delete
       - watch
+  # EndpointSlices are used for Service-based network policy rule
+  # enforcement.
+  - apiGroups: ["discovery.k8s.io"]
```

Comment:
I don't see endpointslices RBAC in [v3.19](https://github.com/projectcalico/calico/blob/release-v3.19/_includes/charts/tigera-operator/templates/tigera-operator/02-role-tigera-operator.yaml) or in [master](https://github.com/projectcalico/calico/blob/master/calico/_includes/charts/tigera-operator/templates/tigera-operator/02-role-tigera-operator.yaml) so I'm wondering why that would be needed.

I also took a look at the operator code and I don't see any list or watch for endpointslices. The only endpointslices references I found in the operator code is for creating the RBAC for giving permissions to node and typha for endpointslices, and that wouldn't require the operator itself to have endpointslices access.

Reply:
@tmjd this role was added in [this PR](https://github.com/aws/amazon-vpc-cni-k8s/pull/1678). For some reason, the manifest wasn't templated in the PR. Can you take a look if this need to be added into Calico code as well? Thanks.

Reply:
That PR was based on another PR ([eks-charts#610](https://github.com/aws/eks-charts/pull/610)) that was adding permissions to calico-node. This PR adds permissions to tigera-operator. You can see that the [operator v1.20.1 does give the same endpointslices permission](https://github.com/tigera/operator/blob/v1.20.1/pkg/render/node.go#L252) to the calico-node role, this is the operator version that the [values file includes](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/charts/aws-calico/values.yaml#L4).

I've commented on the PR you linked that it shouldn't have made any difference.